src/site/content/en/fast/publish-modern-javascript/index.md

```diff
@@ -16,5 +16,383 @@ codelabs:
+
+Use the `"exports"` field along with `"main"` in order to publish your package
+using modern code but also include an ES5 + CommonJS fallback for legacy
+browsers.
+
+```json
+{
+  "name": "foo",
+  "exports": "./modern.js",
+  "main": "./legacy.cjs"
+}
+```
+
+### Modern with legacy fallback and ESM bundler optimizations
+
+In addition to defining a fallback CommonJS entrypoint, the `"module"` field can
+be used to point to a similar legacy fallback bundle, but one that uses
+JavaScript module syntax (`import` and `export`).
+
+```json
+{
+  "name": "foo",
+  "exports": "./modern.js",
+  "main": "./legacy.cjs",
+  "module": "./module.js"
+}
+```
+
+Many bundlers, such as webpack and Rollup, rely on this field to take advantage
+of module features and enable
+[tree shaking](/commonjs-larger-bundles/#how-does-commonjs-affect-your-final-bundle-size).
+This is still a legacy bundle that does not contain any modern code aside from
+`import`/`export` syntax, so use this approach to ship modern code with a
+legacy fallback that is still optimized for bundling.
+
+## Modern JavaScript in Applications
+
+Third-party dependencies make up the vast majority of typical production
+JavaScript code in web applications. While npm dependencies have historically
+been published as legacy ES5 syntax, this is no longer a safe assumption and
+risks dependency updates breaking browser support in your application.
+
+With an increasing number of npm packages moving to modern JavaScript, it's
+important to ensure that the build tooling is set up to handle them. There's a
+good chance some of the npm packages you depend on are already using modern
+language features. There are a number of options available to use modern code
+from npm without breaking your application in older browsers, but the general
+idea is to have the build system transpile dependencies to the same syntax
+target as your source code.
+
+## webpack
+
+As of webpack 5, it is now possible to configure what syntax webpack will use
+when generating code for bundles and modules. This doesn't transpile your code or dependencies, it only affects the "glue" code generated by webpack. The browser support target can be specified by adding a
+[browserslist configuration](https://github.com/browserslist/browserslist#readme)
+to your project, or directly in your webpack configuration:
+
+```js
+module.exports = {
+  target: ['web', 'es2017'],
+};
+```
+
+It is also possible to configure webpack to generate optimized bundles that
+omit unnecessary wrapper functions when targeting a modern ES Modules
+environment. This also configures Webpack to load code-splitted bundles using
+`<script type="module">`.
+
+```js
+module.exports = {
+  target: ['web', 'es2017'],
+  output: {
+    module: true,
+  },
+  experiments: {
+    outputModule: true,
+  },
+};
+```
+
+There are a number of webpack plugins available that make it possible to
+compile and ship modern JavaScript while still supporting legacy browsers.
```

Comment:
```suggestion
There are a number of webpack plugins available that make it possible to
compile and ship modern JavaScript while still supporting legacy browsers,
such as: Optimize Plugin, BabelEsmPlugin, and webpack-plugin-modern-npm.
```

Reply:
I switched the `webpack-plugin-modern-npm` section to be a manual configuration recommendation instead of a plugin to install, so I've tweaked your suggestion here to be:

```md
There are a number of webpack plugins available that make it possible to
compile and ship modern JavaScript while still supporting legacy browsers,
such as Optimize Plugin and BabelEsmPlugin.
```

Hope that's okay!